dts/arm/atmel/saml2x.dtsi

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2021 Argentum Systems Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv6-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include <dt-bindings/pwm/pwm.h>
+
+/ {
+	aliases {
+		watchdog0 = &wdog;
+	};
+
+	chosen {
+		zephyr,flash-controller = &nvmctrl;
+		zephyr,entropy = &trng;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m0+";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 0x8000>;
+	};
+
+	id: device_id@80a00c {
+		compatible = "atmel,sam0-id";
+		reg =	<0x0080A00C 0x4>,
+			<0x0080A040 0x4>,
+			<0x0080A044 0x4>,
+			<0x0080A048 0x4>;
+	};
+
+	aliases {
+		port-a = &porta;
+		port-b = &portb;
+		adc-0 = &adc;
+
+		sercom-0 = &sercom0;
+		sercom-1 = &sercom1;
+		sercom-2 = &sercom2;
+		sercom-3 = &sercom3;
+		sercom-4 = &sercom4;
+		sercom-5 = &sercom5;
+
+		tc-4 = &tc4;
+	};
```

Comment:
Are any of these aliases actually necessary?

Many aliases in zephyr dtsi files are copy/pasted from files that were added before zephyr v2.3. In the old days, we needed aliases because there was no support for accessing nodes by node label. So for example in zephyr v2.2, you could access devicetree data from the `sercom-0` alias, but not the `sercom0` node label. But ever since v2.3, we have DT_NODELABEL() as part of the current devicetree API, so we don't need these sort of redundant aliases anymore as far as I can tell.

Reply:
It seems not...!